### PR TITLE
Session players

### PR DESCRIPTION
**Pull request by jjtfsalgado**
Includes add/remove players from a session

PS: UI needs 💖

### CODE REVIEW

File: backend/Repositories/SessionPlayersRepository.cs [hunk 1]
```diff
@@ -22,24 +23,25 @@ public SessionPlayersRepository(PingPongContext db)
             _db = db;
         }
 
-        public IAsyncEnumerable<SessionPlayer> GetAll()
+        public Task Add(List<SessionPlayer> SessionPlayer)
```
Comment (cri5ti) [inline backend/Repositories/SessionPlayersRepository.cs]: public Task Add(List<SessionPlayer> **sessionPlayers**)

File: backend/Repositories/SessionPlayersRepository.cs [hunk 2]
```diff
@@ -22,24 +23,25 @@ public SessionPlayersRepository(PingPongContext db)
...
-                {
-                    SessionId = i.SessionId,
-                    PlayerId = i.PlayerId
-                })
-                .AsAsyncEnumerable();
+            _db.SessionPlayers.AddRange(SessionPlayer.Select(i => new DbSessionPlayer
+            {
+                SessionId = i.SessionCode.ToGuid(),
+                PlayerId = i.PlayerCode.ToGuid()
+            }).ToList());
```
Comment (cri5ti) [inline backend/Repositories/SessionPlayersRepository.cs]: the ToList() is not necesarry here

File: backend/Repositories/SessionPlayersRepository.cs [hunk 3]
```diff
@@ -22,24 +23,25 @@ public SessionPlayersRepository(PingPongContext db)
...
+            
+            return _db.SaveChangesAsync();
         }
 
-        public Task Add(SessionPlayer SessionPlayer)
+        public Task Delete(List<SessionPlayer> SessionPlayer)
```
Comment (cri5ti) [inline backend/Repositories/SessionPlayersRepository.cs]: same

File: backend/ModelView/SessionPlayer.cs [hunk 4]
```diff
@@ -5,9 +5,8 @@ namespace IP5.Model
 {
 	public class SessionPlayer
 	{
-		public Guid SessionId { get; set; }
-		public Guid PlayerId { get; set; }
-
+		public string SessionCode { get; set; }
+		public string PlayerCode { get; set; }
 		public Session Session { get; set; }
 		public Player Player { get; set; }
```
Comment (cri5ti) [inline backend/ModelView/SessionPlayer.cs]: Isn't the `PlayerCode` the only member used?
Is `SessionCode`, `Session` and `Player` members required / used?

File: frontend/src/api/players.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import {API_URL} from "../config";
 
 
-export type Player = {
+export type TPlayer = {
```
Comment (cri5ti) [inline frontend/src/api/players.ts]: TType naming is weird

File: frontend/src/api/sessions.ts [hunk 6]
```diff
@@ -1,17 +1,39 @@
...
+        method: "DELETE",
+        body: JSON.stringify(players),
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }}),
+    create: async (body: Partial<TSession>): Promise<Response> => await fetch(`${API_URL}/sessions`, {
         method: "PUT",
         body: JSON.stringify(body),
         headers: {
```
Comment (cri5ti) [inline frontend/src/api/sessions.ts]: need to reduce boilerplate code

File: frontend/src/ui/multi_select/multi_select.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        _onChange(_itemsSelected);
+        setState({itemsSelected: _itemsSelected, itemsAvailable: _itemsAvailable, tempAvailableCodes: _tempAvailableCodes, tempSelectedCodes: []})
+    };
+
+    const _onChange = (itemsSelected: Array<T>) => {
+        const added = itemsSelected.filter(i => !initialSelection.some(j => j.code === i.code));
+        const removed = initialSelection.filter(i => !itemsSelected.some(j => j.code === i.code));
+
+        onChange({added, removed})
+    };
```
Comment (cri5ti) [inline frontend/src/ui/multi_select/multi_select.tsx]: useCallback, or class component.
hooks are nice when they improve readability.
here it's not very readable.

you can use hooks and class components in the same project, no biggie.

File: backend/ModelView/SessionPlayer.cs (hunk 4 above)
Comment (jjtfsalgado) [reply backend/ModelView/SessionPlayer.cs]: On put Im using both (PlayerCode & SessionCode) to create the SessionPlayer record, but on get I'm fetching only the Player atm

File: frontend/src/api/sessions.ts (hunk 6 above)
Comment (jjtfsalgado) [reply frontend/src/api/sessions.ts]: I will refactor later on, I will try to create an abstraction of the http client to encapsulate error handling, etc

**Pull request merged**